app/models/company.rb

```diff
@@ -2,4 +2,10 @@ class Company < ActiveRecord::Base
   validates :name, :city, presence: true
   validates :name, uniqueness: true
   has_many :jobs
+  has_many :contacts
+
+  def self.location_job_count
+      Company.joins(:jobs).group(:city).count
```

Comment:
since this is a class method, there's no need to have `Company` here. we could just do `joins(:jobs).group(:city).count`